homeassistant/components/freebox/__init__.py

```diff
@@ -57,23 +75,15 @@ async def async_reboot(call: ServiceCall) -> None:
 
     hass.services.async_register(DOMAIN, SERVICE_REBOOT, async_reboot)
 
-    async def async_close_connection(event):
-        """Close Freebox connection on HA Stop."""
-        await router.close()
-
-    entry.async_on_unload(
-        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_close_connection)
-    )
-
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     if unload_ok:
-        router = hass.data[DOMAIN].pop(entry.unique_id)
-        await router.close()
+        router: FreeboxRouter = hass.data[DOMAIN].pop(entry.unique_id)
+        await router._api.close()  # pylint: disable=[protected-access]
```

Comment:
If we want to keep the api protected we need a non protected router method that closes the api connection.